### ISSUE TITLE
Unreachable code reached when using UNC paths on Windows

### ISSUE THREAD

**Issue by JerwuQu** [ziglang/zig#8205]
Using absolute paths related to network shares (starting with `\\...`) will result in stdlib reaching `unreachable`, meaning it's impossible to recover from. Preferred would be if it spat an error such as `error.UnsupportedDrive` so it can be handled gracefully, or if support for them was added.

Note that getting the working dir handle from `std.fs.cwd()` whilst on a network share **works**, and you _can_ use `openDir`, `openFile`, and all other **relative** methods from there. Meaning if you've somehow gotten a handle, you can access and read/write files. Afaik it only breaks when you try to use absolute paths.

Getting working dir from `std.fs.cwd()` whilst on a network share and then running `realpath` on the handle also breaks with `unreachable`.

I've only experienced this because I've done development in WSL2, which Windows has mounted as a network share.

### Example:
```zig
const std = @import("std");

pub fn main() !void {
    const path = "\\\\wsl$\\Arch\\home\\jerwuqu";
    std.debug.print("Is absolute: {}\n", .{std.fs.path.isAbsolute(path)});

    _ = try std.fs.openDirAbsolute(path, .{});
    std.debug.print("OK!\n", .{});
}
```

#### Output:
```
Is absolute: true
thread 19092 panic: reached unreachable code
Panicked during a panic. Aborting.
```

**Comment by star-tek-mb** [comment]
`std.c.fopen` can open files on WSL, but `std.fs.openFileAbsolute` can't with `\\wsl.localhost\Debian` like path

**Comment by star-tek-mb** [comment]
For files on network share we must use UNC:
```zig
var content = try std.fs.cwd().readFileAlloc(allocator, "\\??\\UNC\\\\wsl.localhost\\Debian\\home\\noobtek\\main.c", std.math.maxInt(usize));
```

For local files UNC also can be used:
```zig
var content = try std.fs.cwd().readFileAlloc(allocator, "\\??\\UNC\\localhost\\D$\\main.c", std.math.maxInt(usize));
```

`\\??\\UNC\\localhost\\D:\\main.c` - doesn't work.

So, I think we need to handle these edge cases, at least on std.debug on stack traces.

**Comment by andrewrk** [comment]
Fix reverted in b5fad3a40a86eb379903d6a803bdbe66dcaa5487 due to master branch CI failures

**Comment by squeek502** [comment]
This can be re-closed since the fix was unreverted in https://github.com/ziglang/zig/pull/15905
